torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -38,38 +38,7 @@ C10_EXPORT std::vector<at::DeprecatedTypeProperties*> allCUDATypes() {
   return allTypesForBackends({ Backend::CUDA, Backend::SparseCUDA });
 }
 
-namespace {
-const Variable & checked_cast_variable(const Tensor & t, const char * name, int pos) {
-  if (!t.defined()) {
-    AT_ERROR("Expected a Tensor of type Variable but found an undefined Tensor for argument #", pos, " '", name, "'");
-  }
-  return t;
-}
-
-Variable & checked_cast_variable(Tensor & t, const char * name, int pos) {
-  if (!t.defined()) {
-    AT_ERROR("Expected a Tensor of type Variable but found an undefined Tensor for argument #", pos, " '", name, "'");
-  }
-  return t;
-}
-}
-
-const Tensor & unpack(const Tensor & t, const char * name, int pos) {
```

Comment:
Are these functions part of the c++ API? Is this BC-breaking?

Reply:
Not to my knowledge. (@ezyang?) If they are, we will still want to make them call the macros and directly use the macros in our own code for perf.